packages/markers/src/browser/marker-manager.ts

```diff
@@ -160,6 +172,22 @@ export abstract class MarkerManager<D extends object> {
         return oldMarkers;
     }
 
+    /*
+     * adds markers for the given uri and owner with the given data, without touching the exsting markers associated with the same uri and owner.
+     */
+    addMarkers(uri: URI, owner: string, data: D[]): Marker<D>[] {
+        const uriString = uri.toString();
+        const collection = this.uri2MarkerCollection.get(uriString) || new MarkerCollection<D>(uri, this.getKind());
+        const newMarkers = collection.addMarkers(owner, data);
```

Comment:
Also code looks strange why adding will delete something? Do you have tests for new APIs?

Reply:
Yes I shouldn't have the "delete". Will add tests.